DataDog/serverless-plugin-datadog
# Update SLS plugin to support"forwarder" or "forwarderArn" parameters

### Update SLS plugin to support"forwarder" or "forwarderArn" parameters


### What does this PR do?
This PR updates the Serverless Plugin "forwarder" parameter to "forwarderArn" to match the configuration settings of the Datadog CloudFormation Macro while still keeping support for the "forwarder" parameter in efforts of not breaking previously instrumented Serverless functions. This PR also updates the Serverless Plugin README to document the capability to use either "forwarder" or "forwarderArn" parameters.


### Motivation
My motivation for this PR is that the [Datadog Serverless Plugin](https://github.com/DataDog/serverless-plugin-datadog) used the "forwarder" parameter while the [Datadog CloudFormation Macro](https://github.com/DataDog/datadog-cloudformation-macro/tree/master/serverless) used the "forwarderArn" parameter, "forwarder" and "forwarderArn" are synonymous as they are ultimately configured to take in the Datadog forwarder function arn. Updating "forwarder" to "forwarderArn" keeps the convention consistent between the SLS plugin and CFM macro and hopefully eliminates any previous confusion.


### Testing Guidelines
I created unit tests in the index.spec.ts file covering the use of "forwarderArn" and "forwarder" parameters in the Serverless.yml file which confirmed expected results through the presence of a created subscription filter. I also manually configured a test Serverless function with the use of "forwarderArn" and "forwarder" and confirmed the results in the AWS console (CloudWatch Log groups).


### Additional Notes
I had a local issue running the index.spec.ts file but this problem was persistent on the master branch as well as my development branch so it is most likely a machine problem.


### Types of changes


### Check all that apply
